interface/src/avatar/AvatarMotionState.cpp

```
@@ -172,7 +172,7 @@ QUuid AvatarMotionState::getSimulatorID() const {
 // virtual
 void AvatarMotionState::computeCollisionGroupAndMask(int32_t& group, int32_t& mask) const {
     group = _collisionGroup;
-    mask = _collisionGroup == BULLET_COLLISION_GROUP_COLLISIONLESS ? 0 : Physics::getDefaultCollisionMask(group);
+    mask = _avatar->getCollideWithOtherAvatars() ? BULLET_COLLISION_MASK_DEFAULT : 0;
```

This is not correct.  It should be something like this:
```
mask = Physics::getDefaultCollisionMask(group);
if (!_avatar->getCollideWithOtherAvatars()) {
    // remove all AVATAR bits from mask
    mask &= ~(BULLET_COLLISION_GROUP_MY_AVATAR | BULLET_COLLISION_GROUP_OTHER_AVATAR);
}
```
Note: the default collision mask corresponding to `BULLET_COLLISION_GROUP_OTHER_AVATAR` does not include the `BULLET_COLLISION_GROUP_OTHER_AVATAR` bit so removing it here is a **no-op**, however I still think we should visibly remove it here because it makes the **intent** of the code more clear to future devs.

The only valid measure of code quality is the [wtfs/min](http://www.osnews.com/images/comics/wtfm.jpg).